File: src/main/java/com/wx/java_code/sort/SelectionSort.java
package com.wx.java_code.sort;

public class SelectionSort {


    public static void main(String[] args) {
        int[] array = new int[] {3,4,5,1,2,4,6,7,8};
        selectionSort(array);
        System.out.println("[ ");
        for (int a : array) {
            System.out.print(a+" ");
        }
        System.out.print(" ]");
    }

    public static void selectionSort(int[] array) {
        for (int i = 0; i < array.length; i++){
            int min = i;
            for (int j = i + 1; j < array.length; j++) {
                if (array[j] < array[min]) {
                    min = j;
                }
            }
            if (min != i) {
                int temp = array[i];
                array[i] = array[min];
                array[min] = temp;
            }
        }
    }
}

File: src/main/java/com/wx/kotlin_code/sort/SelectionSort.kt
package com.wx.kotlin_code.sort


/**
 * 选择排序算法
 **/
class SelectionSort <T> : BaseSort() {

    fun selectSort(array: Array<T>) {
        show("排序前：", array)
        for (i in array.indices) {
            var min = i
            for(j in i + 1 until  array.size) {
                if (less(array[j], array[min])) {
                    min = j
                }
                if (min != i) {
                    exch(array, min, j)
                }
            }
        }
        show("排序后：", array)
    }

}

fun main() {
    val selectionSort = SelectionSort<Int>()
    val data = arrayOf(3,4,5,1,2,4,6,7,8)
    selectionSort.selectSort(data)
}